deploy/config/deploy/yum.rb

```diff
@@ -63,4 +68,10 @@
       run "if [ -e #{repo_instance_path}-previous ]; then rm -rf #{repo_instance_path}-previous; fi" if overwrite || merge
     end
   end
+
+  task :purgecdn do
+    run %Q{awk '/ CHANGED /{print "https://yum.theforeman.org/"#{repo_dest}$5}' #{repo_rsync_log} | xargs --no-run-if-empty curl -X PURGE -H 'Fastly-Soft-Purge:1'}
```

Comment:
I think the script can be called with `replace` instead of `rsync` in which case the file will not exist. Probably good to check if it exists.

Reply:
not sure what you mean. there is a `replace` task, but if that is called, the `purgecdn` task is not called either. and if one calls the `sync` task, the `rsync` task is always executed.

Reply:
So I was mixing things up and confused. Tasks are always executed but we do have parameters to tune:

https://github.com/theforeman/foreman-infra/blob/2cc59353f8d74943e687fac2724967e2b40176e8/puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/release.groovy#L12